File: practicals/practicals/LABS/ConsoleApp1/MyString.cs
using System;
using System.Linq;

public class MyString
{
    public string Value { get; set; }

    public MyString(string value)
    {
        Value = value;
    }

    public int GetLength()
    {
        return Value.Length;
    }

    public string SortDescending()
    {
        return new string(Value.OrderByDescending(c => c).ToArray());
    }

    public override string ToString()
    {
        return Value;
    }
}

File: practicals/practicals/LABS/C++/MyString.cpp
#include "MyString.h"
#include <iostream>

MyString::MyString(std::string value) : value(value) {}

int MyString::getLength() {
    return value.length();
}

std::string MyString::sortDescending() {
    std::string sorted = value;
    std::sort(sorted.rbegin(), sorted.rend());
    return sorted;
}

std::ostream& operator<<(std::ostream& os, const MyString& myString) {
    os << myString.value;
    return os;
}

